File: app/src/main/java/com/example/projektkomp/MainActivity.java
package com.example.projektkomp;

import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void openTreningView(View view)
    {
        setContentView(R.layout.activity_trening_main);
        // Do something in response to button click
    }

    public void openOrganizerView(View view)
    {
        setContentView(R.layout.activity_organizer_main);
        // Do something in response to button click
    }

    public void openRaportView(View view)
    {
        setContentView(R.layout.activity_raport_main);
        // Do something in response to button click
    }

    public void backToPrevView(View view)
    {
        setContentView(R.layout.activity_main);
        // Do something in response to button click
    }

    ///////////////////////////////////////

    public void openDietView(View view)
    {
        setContentView(R.layout.activity_trening_dieta);
    }

}

File: app/src/main/java/com/example/application/view/activity/MainActivity.kt
package com.example.application.view.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.core.view.GravityCompat
import androidx.navigation.findNavController
import com.example.application.R
import com.example.application.view.customView.TopBar
import com.example.application.view.navigation.NavigationAdapter
import com.example.application.viewModel.MainActivityViewModel
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), CoreFragmentInterface {

    private val viewModel by lazy { MainActivityViewModel() }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        viewModel.deleteSession()

        navigation.setAdapter(NavigationAdapter(drawerLayout, findNavController(R.id.main_host_fragment)), viewModel.menuItems)

        topBar.onMenuClick = {
            openDrawer()
        }
    }

    private fun openDrawer(){
        drawerLayout.openDrawer(GravityCompat.START, true)
    }

    override fun getTopBar(): TopBar = topBar
}
